File: lib/taskana-core/src/main/java/pro/taskana/task/api/TaskState.java
package pro.taskana.task.api;

import java.util.Arrays;

/** The TaskState contains all status of a {@linkplain pro.taskana.task.api.models.Task Task}. */
public enum TaskState {
  READY,
  CLAIMED,
  READY_FOR_REVIEW,
  IN_REVIEW,
  COMPLETED,
  CANCELLED,
  TERMINATED;

  public static final TaskState[] END_STATES = {COMPLETED, CANCELLED, TERMINATED};

  public boolean in(TaskState... states) {
    return Arrays.asList(states).contains(this);
  }

  public boolean isEndState() {
    return in(END_STATES);
  }
}

File: web/src/app/shared/models/task-state.ts
export enum TaskState {
  ALL = 'ALL',
  READY = 'READY',
  CLAIMED = 'CLAIMED',
  COMPLETED = 'COMPLETED',
  CANCELLED = 'CANCELLED',
  TERMINATED = 'TERMINATED'
}

export const ALL_STATES: Map<TaskState, string> = new Map([
  [TaskState.ALL, 'All'],
  [TaskState.READY, 'Ready'],
  [TaskState.CLAIMED, 'Claimed'],
  [TaskState.COMPLETED, 'Completed'],
  [TaskState.CANCELLED, 'Cancelled'],
  [TaskState.TERMINATED, 'Terminated']
]);
